### PR TITLE
Added pyramid aggregation methods

### LINKED ISSUE #655: Generate pyramids using given aggregation method
**Is your feature request related to a problem? Please describe.**

We currently generate multi-level datasets by pixel selection only.
Although the method keeps original measurements, the visual result for higher levels is often really bad for pyramids with 4 or more levels. 

**Describe the solution you'd like**

Allow using [`xarray.DataArray.coarsen`](https://xarray.pydata.org/en/stable/generated/xarray.DataArray.coarsen.html) with given aggregation method, namely min, max, median, mean.

Hence, add new write parameter `agg_method` to file data stores with possible values `None` (= pixel selection, the default), `"min"`, `"max"`, `"mean"`, or `"median"`.

### PR DESCRIPTION
**Pull request by forman**
Closes #655

In detail:

- Introduced new parameter `agg_methods` for writing multi-level datasets with the "file", "s3", and "memory" data stores. The value of `agg_methods` is either a string `"first"`, `"min"`, `"max"`, `"mean"`, `"median"` or a dictionary that maps a variable name to an aggregation method. Variable names can be patterns that may contain wildcard characters '*' and '?'. The special aggregation method `"auto"` can be used to select `"first"` for integer variables and `"mean"` for floating point variables. 

- The `xcube level` CLI tool now has a new option `--agg-method` (or `-A`) for the same purpose.

Checklist:

* [x] Add unit tests and/or doctests in docstrings
* [x] Add docstrings and API docs for any new/modified user-facing classes and functions
* [x] New/modified features documented in `docs/source/*`
* [x] Changes documented in `CHANGES.md`
* [x] AppVeyor CI passes
* [x] Test coverage remains or increases (target 100%)

### CODE REVIEW

File: test/core/test_subsampling.py [hunk 1]
```diff
@@ -25,56 +25,237 @@
...
-        self.assertIn('grid_mapping',
-                      subsampled_dataset.test_var.attrs)
+
+        self.assertIn('x',
+                      subsampled_dataset.coords)
+        self.assertIn('y',
+                      subsampled_dataset.coords)
+
+        print(repr(subsampled_dataset.x.values))
+        print(repr(subsampled_dataset.y.values))
```
Comment (TonioF) [inline test/core/test_subsampling.py:204]: Why print them out?
```suggestion
```

File: test/core/test_subsampling.py [hunk 2]
```diff
@@ -25,56 +25,237 @@
...
+        self.assertIn('grid_mapping',
+                      subsampled_dataset.var_2.attrs)
+
+        self.assertEqual(expected_var_1.dtype,
+                         subsampled_dataset.var_1.dtype)
+        self.assertEqual(expected_var_2.dtype,
+                         subsampled_dataset.var_2.dtype)
+
+        # print(repr(subsampled_dataset.var_1.values))
+        # print(repr(subsampled_dataset.var_2.values))
```
Comment (TonioF) [inline test/core/test_subsampling.py:231]:
```suggestion
```
Same here.

File: test/core/test_subsampling.py (hunk 1 above)
Comment (forman) [reply test/core/test_subsampling.py:204]: Forget to remove. Will fix.

**Review by thomasstorm — CHANGES_REQUESTED** [review]
See suggestions

**Review by thomasstorm — APPROVED** [review]

**Pull request merged**